src/units/attack_type.hpp

```diff
@@ -152,6 +152,13 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	 * @param from unit adjacent to self_ is checked.
 	 */
 	bool check_adj_abilities(const config& cfg, const std::string& special, int dir, const unit& from) const;
+	/** anti_special_or_ability : return an boolean value if detect ability/specials to overwrite
+	 * @return True if the special @a special matches.
+	 * @param special the config to one special ability checked.
+	 * @param type The special ability type who is being checked.
+	 * @param whom If special checked applied to self or other.
+	 */
+	bool anti_special_or_ability(const config& special , const std::string& type, AFFECTS whom) const;
```

Comment:
Please add to this documentation *why* it chooses to overwrite a special. Saying that something is "being checked" doesn't say what the check is.

Reply:
it' done